### DISCUSSION TITLE
Single quotes added when creating interfaces on module

### DISCUSSION THREAD

**Discussion by jguiles06** [netbox-community/netbox#17607] [General]
In Netbox when creating a new module type and adding interfaces, single quotes are added when the {module} variable is used. For example:

xe-0/{module}/0 results in xe-0/'0'/0 when adding the module to a device.

Is there a way to prevent this from happening or is my only option to strip the single quotes out when rendering a template?

**Comment by candlerb** [comment]
I've never seen that. How to reproduce? What Netbox version are you using?

**Reply by jguiles06** [reply]
Netbox 4.1.1

I have defined an LMIC module for a Juniper MX304 with interfaces defined as in the below screenshot:

<img width="280" alt="Screenshot 2024-09-25 at 5 30 07 PM" src="https://github.com/user-attachments/assets/6ac524f2-2a8b-44c0-bdd0-d43c4eea60ea">

When this module is inserted into the device the resulting interfaces look like this:

<img width="191" alt="Screenshot 2024-09-25 at 5 29 21 PM" src="https://github.com/user-attachments/assets/7168a21e-56bc-46db-9ec8-257a3f75ed06">


Happy to provide any other info just let me know what you need.

**Reply by candlerb** [reply]
I have just tested it on demo.netbox.dev, which is running 4.1.1, and it works fine there. (Note: I had to turn off the broken "HTMX Navigation" feature so I could browse the module interfaces, and also the language had been set to Portuguese so I turned it back to Auto)

On device "ncsu-coreswitch1" I added a module of type "EX9200-32XS" into bay 2, and all the newly created interfaces are correctly numbered without single quotes:

![image](https://github.com/user-attachments/assets/7aeef015-b881-46ed-b958-00bb4a98f470)

Therefore, either you're doing something different to what I'm doing, or there's something weird about your system:
* if HTMX Navigation is turned on, have you tried turning it off?
* are you using language translations in the UI? It's unlikely, but not impossible, that it breaks things.
* can you try with a different web browser? I'm using Chrome under macOS 14.7.
* maybe Netbox is installed incorrectly? See if you can replicate your problem on demo.netbox.dev, and that should make it clear whether the problem is with the Netbox server, or the client environment.

------

I have also tested it with my local Netbox 4.0.11 and it works fine. I created a module type, and added 8 interface templates in one go, using name pattern `xe0/{module}/1:[0-7]` to be as close to yours as possible.

Here's the module bay that I added to a random device:

![image](https://github.com/user-attachments/assets/bd6e0da0-6267-4e40-a3d6-2c2a315bf8e2)

And after inserting the module into that module bay, these are the interfaces on the device:

![image](https://github.com/user-attachments/assets/320c5dfc-6357-4bb3-a7bb-47744d82eaaf)

**Reply by candlerb** [reply]
> I had to turn off the broken "HTMX Navigation" feature so I could browse the module interfaces

Just to clarify that remark: with HTMX Navigation turned on, when I was at https://demo.netbox.dev/dcim/module-types/1/ and tried to click the "Interfaces" tab:

![image](https://github.com/user-attachments/assets/8476c98b-80da-44ef-a883-ba64e6c7e95f)

Nothing happened in the UI. The developer console showed a 500 server error:

![image](https://github.com/user-attachments/assets/01be68ae-7e3a-41b9-85ea-732d329731fa)

And the following response body:

![image](https://github.com/user-attachments/assets/a4505aa7-b0b8-4384-9a4c-168b310fed65)

Since the default appears to be "disabled" I wonder if some other user had turned it on, as well as switching the language to Portuguese.

EDIT: I believe the HTMX problem is issue #17360 (that was reported for interfaces on a device, rather than interface templates on a module type) and the whole HTMX strategy is under review in #17571.

**Reply by jguiles06** [reply]
Okay...so this was a problem entirely of my own creation. 🤦🏼‍♂️ For some unknown reason when I created the base device I put single quotes around the numbers in the position field for the module bays. 

<img width="614" alt="Screenshot 2024-09-26 at 3 59 23 PM" src="https://github.com/user-attachments/assets/009e821c-08ad-4b9c-8dd3-f7b0078edf9e">

I've been sitting here trying every permutation possible and kept getting the same result. I assumed it was something with the configuration of the module so it took me a while to check the device itself. 

I appreciate you taking a look at this and apologize for wasting your time.

**Comment by jguiles06** [comment]
Wow! Thank you!! I'm going to try your suggestions and probably create some new modules to see if there is something I'm doing incorrectly. I'll report back with my findings.
